test/api_integration/apis/data_views/has_user_index_pattern/has_user_index_pattern.ts

```diff
@@ -33,7 +37,11 @@ export default function ({ getService }: FtrProviderContext) {
         const servicePath = `${config.basePath}/has_user_${config.serviceKey}`;
 
         it('should return false if no index patterns', async () => {
-          const response = await supertest.get(servicePath);
+          // Make sure all saved objects including data views are cleared
+          await esArchiver.emptyKibanaIndex();
```

Comment:
This part is the fix -- sometimes on subsequent runs the Kibana index doesn't get completely cleared and there are saved objects remaining. I looked into why this was (even thought at one point it might be related to usage collection), but I wasn't able to nail down what was causing it, just that this fixes the issue by making sure it's cleaned before checking.